### PR TITLE
Add missing architectures, change deployment target to iOS 10

### PR DESCRIPTION
**Pull request by jackpooleybc**
This PR mainly fixes two issues:
* It changes the `Deployment Target` to `iOS 10.0`, and the Swift language version `Swift 5` in the `podspec`
* It adds missing architectures to the `libwally-core` build script

### CODE REVIEW

**Comment by Sjors** [comment]
iOs 10 indeed still supports the 32 bit A6 chip. This should be tested on an iPhone 5(C). See https://i.stack.imgur.com/jKZE4.png

Why are you dropping `-fembed-bitcode`? (I have no idea what it does)

File: build-libwally.sh [hunk 1]
```diff
@@ -50,8 +50,8 @@ if [ $simulator == 1 ]; then
   if [ ! -d "build" ]; then
     echo "Configure and compile for the simulator..."
     set -v
-    export CFLAGS="-O3 -arch x86_64 -arch i386 -mios-simulator-version-min=10.0 -isysroot `xcrun -sdk iphonesimulator --show-sdk-path`"
-    export CXXFLAGS="-O3 -arch x86_64 -arch i386 -mios-simulator-version-min=10.0 -isysroot `xcrun -sdk iphonesimulator --show-sdk-path`"
+    export CFLAGS="-O3 -arch x86_64 -arch i386 -fembed-bitcode-marker -mios-simulator-version-min=10.0 -isysroot `xcrun -sdk iphonesimulator --show-sdk-path`"
```
Comment (Sjors) [inline build-libwally.sh]: Maybe squash this into the first commit where it's removed?

**Comment by Sjors** [comment]
Can you drop a29fe5b2a1e3bc53ba325e5152bd39ff1f2547bd? I merged it from #12 because that also updated the README.

**Comment by jackpooleybc** [comment]
@Sjors done

**Comment by Sjors** [comment]
Merge using https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/github-merge.py

With the commands:

```sh
git config githubmerge.repository blockchain/libwally-swift
../bitcoin-maintainer-tools/github-merge.py 11
```

A had to make a small tweak to the merge script to handle the submodule, so it might be incorrect...

**Pull request merged**